internal/action/actions.go

```diff
@@ -175,6 +175,9 @@ func (h *BufPane) MoveCursorUp(n int) {
 			vloc.SLoc = sloc
 			vloc.VisualX = h.Cursor.LastVisualX
 			h.Cursor.Loc = h.LocFromVLoc(vloc)
+			if util.IntOpt(h.Buf.Settings["wrapindent"]) > -1 {
```

Comment:
It seems like the whole thing is working without this, and this causes the bug I mentioned moving from the first line to the second line (2.)

Reply:
Lovely, indeed it does, great stuff, and much cleaner code to boot :) Have pushed a commit without the AdjustXCursorOnWrapIndent stuff. Have tried out all the other edge cases and they seem to work too at this end.